com.unity.shadergraph/Editor/Data/Nodes/Math/Matrix/MatrixSwizzleNode.cs

```diff
@@ -0,0 +1,540 @@
+                    break;
+                case SwizzleOutputSize.Vector2:
+                    AddSlot(new Vector2MaterialSlot(OutputSlotId, kOutputSlotName, kOutputSlotName, SlotType.Output, Vector2.zero));
+                    break;
+                case SwizzleOutputSize.Vector1:
+                    AddSlot(new Vector1MaterialSlot(OutputSlotId, kOutputSlotName, kOutputSlotName, SlotType.Output, 0));
+                    break;
+                default:
+                    AddSlot(new Matrix4MaterialSlot(OutputSlotId, kOutputSlotName, kOutputSlotName, SlotType.Output));
+                    break;
+            }
+            RemoveSlotsNameNotMatching(new int[] { InputSlotId, OutputSlotId });
+        }
+
+        int[] getIndex(float input)
+        {
+            var row = (int)input;
+            float temp_col = (float)(input - Math.Truncate(input)) * 10;
+
+            int[] index = { row, (int)temp_col };
+            //Debug.Log("row:"+index[0]+ ", col:"+ temp_col);
+            return index;
+        }
+
+        string mapComp(int n)
+        {
+            switch (n)
+            {
+                default:
+                    return "r";
+                case 1:
+                    return "g";
+                case 2:
+                    return "b";
+                case 3:
+                    return "a";
+            }
+        }
+
+
+        Vector4  StringToVec4 (string value)
+        {
+            char[] value_char = value.ToCharArray();
+
+
+            if (value.Length == 8)
+            {
+                for (int i = 0; i < value.Length; i++)
+                {
+                    if (value[i] < '0' || value[i] > '3')
+                    {
+                        value_char[i] = '0';
+                        AreIndiciesValid = false;
+                        ValidateNode();
+                    }
+                }
+
+                float x0 = (float)Char.GetNumericValue(value_char[0]);
+                float x1 = (float)Char.GetNumericValue(value_char[1]);
+                float y0 = (float)Char.GetNumericValue(value_char[2]);
+                float y1 = (float)Char.GetNumericValue(value_char[3]);
+                float z0 = (float)Char.GetNumericValue(value_char[4]);
+                float z1 = (float)Char.GetNumericValue(value_char[5]);
+                float w0 = (float)Char.GetNumericValue(value_char[6]);
+                float w1 = (float)Char.GetNumericValue(value_char[7]);
+
+                float x = x0 + x1 * 0.1f;
+                float y = y0 + y1 * 0.1f;
+                float z = z0 + z1 * 0.1f;
+                float w = w0 + w1 * 0.1f;
+                return new Vector4(x, y, z, w);
+            }
+            else
+            {
+                AreIndiciesValid = false;
+                ValidateNode();
+            }
+            return new Vector4(0, 0, 0, 0);
+        }
+
+        private bool AreIndiciesValid = true;
```

Comment:
You shouldn't put member variables in between functions like this, this should be at the top with the other member variables. Also, the spelling needs to be fixed.